File: Data Structures & Algorithms/two-integer-sum/submission-3.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        answer_map = {}
        result_list = []
        for i in range(len(nums)):
            if (target - nums[i]) in answer_map:
                result_list.append(answer_map[target - nums[i]])
                result_list.append(i)
                
            answer_map[nums[i]] = i
        return result_list

File: Data Structures & Algorithms/missing-number/submission-3.java
class Solution {
    public int missingNumber(int[] nums) {
        int n = nums.length;
        for (int i = 0; i < n; i++) {
            while (nums[i] != i && nums[i] != n) {
                swap(nums, nums[i], i);
            }
        }

        for (int i = 0; i < n; i++) {
            if (nums[i] != i) {
                return i;
            }
        }
        return n;
    }

    private static void swap(int[] nums, int i, int j) {
        int temp = nums[i];
        nums[i] = nums[j];
        nums[j] = temp;
    }
}